src/main/java/com/danielflower/apprunner/runners/GoRunner.java

```diff
@@ -0,0 +1,70 @@
+package com.danielflower.apprunner.runners;
+
+import com.danielflower.apprunner.problems.ProjectCannotStartException;
+import org.apache.commons.exec.CommandLine;
+import org.apache.commons.exec.ExecuteWatchdog;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.maven.shared.invoker.InvocationOutputHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class GoRunner implements AppRunner {
+    public static final Logger log = LoggerFactory.getLogger(GoRunner.class);
+
+    public static final String[] startCommands = new String[]{"go build", "go test", "./{app_dir_name}"};
+    private final File projectRoot;
+    private final String goPath;
+    private final CommandLineProvider goCmd;
+    private ExecuteWatchdog watchDog;
+
+    public GoRunner(File projectRoot, String goPath, CommandLineProvider goCmd) {
+        this.projectRoot = projectRoot;
+        this.goPath = goPath;
+        this.goCmd = goCmd;
+    }
+
+    @Override
+    public File getInstanceDir() {
+        return projectRoot;
+    }
+
+    public void start(InvocationOutputHandler buildLogHandler, InvocationOutputHandler consoleLogHandler, Map<String, String> envVarsForApp, Waiter startupWaiter) throws ProjectCannotStartException {
+        envVarsForApp.put("GOPATH", goPath);
+        try {
+            rungo(buildLogHandler, envVarsForApp, "get");
```

Comment:
Yeah, I think having one workspace makes sense while you are working on projects that are related to each other. For AppRunner, we really really want independence between apps.

Having said that, maybe sharing the workspace is okay, but in that case the app's instance dir should be in the workspace, which it's not, right?